spec/serializers/v3/product_serializer_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe V3::ProductSerializer do
+  let(:sled15) { create(:product, name: 'SLED') }
+  let(:sles15) { create(:product, name: 'SLES') }
+  let!(:basesystem) do
+    create(:product, :extension, name: 'BASESYSTEM', base_products: [sles15, sled15], recommended: true)
+  end
+  let!(:server_applications) do
+    create(:product, :extension, name: 'SERVER APPLICATIONS', base_products: [basesystem], root_product: sles15)
+  end
+  let(:base_url) { 'http://example.com' }
+
+  describe 'SLES extension tree' do
+    subject(:serializer) { described_class.new(sles15, root_product: sles15, base_url: base_url) }
+
+    let(:top_extension) { serializer.as_json[:extensions].first }
+    let(:nested_extension) { top_extension[:extensions].first }
+
+    it 'has base system extension' do
+      expect(top_extension[:name]).to eq(basesystem.name)
+    end
```

Comment:
These can be re-written as:
`its([:name]) { is_expected.to eq basesystem.name }`

Same for the tests below.

Reply:
Not really, it's not the `subject` of the spec.

For the top level specs I'd need to do:

```
subject { described_class.new(sled15, root_product: sled15, base_url: base_url)[:extensions].first }
```

And for nested extensions I'd need to do:

```
    subject { described_class.new(sled15, root_product: sled15, base_url: base_url)[:extensions].first[:extensions].first }
```

That doesn't seem to make anything better, though.